src/Tools/ExternalAccess/Razor/Cohost/Handlers/DocumentSymbols.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Roslyn.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using LSP = Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Cohost.Handlers;
+
+internal static class DocumentSymbols
+{
+    public static async Task<SumType<DocumentSymbol[], SymbolInformation[]>> GetDocumentSymbolsAsync(Document document, bool useHierarchicalSymbols, CancellationToken cancellationToken)
+    {
+        // The symbol information service in Roslyn lives in EditorFeatures and has VS dependencies. for glyph images,
+        // so isn't available in OOP. The default implementation is available in OOP, but not in the Roslyn MEF composition,
```

Comment:
> but not in the Roslyn MEF composition

Hmm - not sure why this isn't the case.  I think its regular workspace service.  Not 100% sure but I would have thought it should be available.

Reply:
oh is the protocol project not in the OOP mef composition?

Reply:
Yes, precisely. That might come up again in other endpoints so could be something we need to talk about in future, but document symbol didn't seem important enough to block on it